src/utils/thread_helper.cpp

```diff
@@ -369,6 +369,12 @@ void ThreadHelper::Scan(ScanHandler aHandler)
     }
 }
 
+void ThreadHelper::SetDhcp6PdStateCallback(Dhcp6PdStateCallback aCallback)
+{
+    mDhcp6PdCallback = aCallback;
+    otBorderRoutingDhcp6PdSetRequestCallback(mInstance, &ThreadHelper::BorderRoutingDhcp6PdCallback, this);
+}
+
```

Comment:
How about moving this definition before the `BorderRoutingDhcp6PdCallback` definitions so all related methods are close to each other?